include/rocksdb/table.h

```diff
@@ -669,6 +683,10 @@ struct BlockBasedTableOptions {
   //
   // Default: 2
   uint64_t num_file_reads_for_auto_readahead = 2;
+
+  // Policy on whether to prepopulate blocks generated by compaction.
+  // Only apply when `prepopulate_block_cache=kFlushAndCompaction`
+  std::shared_ptr<CompactionPrepulateBlockCachePolicy> compaction_prepoulate_block_cache_policy{nullptr};
```

Comment:
Nit: you can just use `std::function<...>`, like this: https://github.com/rockset/rocksdb-cloud/blob/9c202a6984842670da602dff23bc08fd34e278d4/include/rocksdb/options.h#L1901